forum/views.py

```diff
@@ -1,3 +1,74 @@
-from django.shortcuts import render
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+from django.views import View
+from django.shortcuts import get_object_or_404, redirect
 
-# Create your views here.
+from .models import (
+    Forum,
+    Post,
+    Comment,
+    CommentVote,
+    UPVOTE,
+    DOWNVOTE
+)
+
+
+class ForumDetailView(DetailView):
+    model = Forum
+
+
+class ForumListView(ListView):
+    model = Forum
+
+
+class PostDetailView(DetailView):
+    model = Post
+    pk_url_kwarg = 'post_id'
+
+
+class CommentDetailView(DetailView):
+    model = Comment
+    pk_url_kwarg = 'comment_id'
+
+
+class CommentUpvoteView(LoginRequiredMixin, View):
+    def post(self, request, *args, **kwargs):
+        comment = get_object_or_404(Comment, id=self.kwargs.get('pk'))
+
+        try:
+            comment_vote = CommentVote.objects.get(
+                user=self.request.user,
+                comment=comment
+            )
+            comment_vote.vote = UPVOTE
+            comment_vote.save()
+        except CommentVote.DoesNotExist:
+            _ = CommentVote.objects.create(
```

Comment:
line 46 uses a built in function/attribute that is equivalent to (if object == null) or if query set for models does not exist right?  
site....
"This exception is raised by the ORM in a couple places, for example by QuerySet.get() when an object is not found for the given query parameters.
Django provides a DoesNotExist exception as an attribute of each model class to identify the class of object that could not be found and to allow you to catch a particular model class with try/except.""